plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieTableOperations.java

```diff
@@ -126,7 +128,11 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
     {
         verify(version.orElseThrow() >= 0, "commitToExistingTable called on a new table");
         try {
-            nessieClient.commitTable(base, metadata, writeNewMetadata(metadata, version.getAsInt() + 1), table, toKey(new SchemaTableName(database, this.tableName)));
+            if (table == null) {
+                table = nessieClient.table(toIdentifier(new SchemaTableName(database, tableName)));
```

Comment:
it's not clear to me why these changes are necessary

Reply:
`nessieClient.commitTable` signature has changed. It just needs contentID instead of expected content. 

`ExpectedContent`(aka table)  was null before this PR also for `commitToExistingTable`. I have debug the original integration to understand it.